catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/ChangeVariableTest.java

```diff
@@ -0,0 +1,136 @@
+import static android.support.test.espresso.assertion.ViewAssertions.doesNotExist;
+import static android.support.test.espresso.matcher.ViewMatchers.hasSibling;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickDataInteractionWrapper.onBrickAtPosition;
+import static org.hamcrest.Matchers.allOf;
+
+@RunWith(AndroidJUnit4.class)
+public class ChangeVariableTest {
+	private ChangeVariableBrick changeVariableBrick;
+
+	@Rule
+	public BaseActivityInstrumentationRule<ScriptActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(ScriptActivity.class, true, false);
+
+	@Before
+	public void setUp() throws Exception {
+		createProject();
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@Test
+	public void testCreateNewUserVariableAndDeletion() {
+		String userVariableName = "testVariable1";
+		String secondUserVariableName = "testVariable2";
+
+		onBrickAtPosition(0)
+				.checkShowsText(R.string.brick_when_started);
+		onBrickAtPosition(1)
+				.checkShowsText(R.string.brick_change_variable);
+		onBrickAtPosition(1).onVariableSpinner(R.id.change_variable_spinner)
+				.performNewVariableInitial(userVariableName);
+
+		// fails due to CAT-2359
+		onBrickAtPosition(1).onVariableSpinner(R.id.change_variable_spinner)
+				.performNewVariable(secondUserVariableName);
+
+		onBrickAtPosition(1).onVariableSpinner(R.id.change_variable_spinner)
+				.checkHasVariable(userVariableName);
+		onBrickAtPosition(1).onVariableSpinner(R.id.change_variable_spinner)
+				.checkHasVariable(secondUserVariableName);
+
+
+		onBrickAtPosition(1).onFormulaTextField(R.id.brick_change_variable_edit_text)
+				.perform(click());
+		BrickTestUtils.deleteItemWithTextFromFormulaEditorDataList(userVariableName);
+
+		onView(allOf(withText(userVariableName), hasSibling(withText("0.0"))))
+				.check(doesNotExist());
+		pressBack();
+		onView(withText(userVariableName))
+				.check(doesNotExist());
+	}
+
+	@Test
+	public void testCreateUserVariableInFormulaEditor() {
+		String userVariableName1 = "Variable1";
+
+		onBrickAtPosition(1).onFormulaTextField(R.id.brick_change_variable_edit_text)
+				.performNewVariable(userVariableName1);
+
+		onBrickAtPosition(1)
+				.checkShowsText(userVariableName1);
+
+		onBrickAtPosition(1).onVariableSpinner(R.id.change_variable_spinner)
+				.checkHasVariable(userVariableName1);
+	}
+
+	@Test
+	public void testViewInFormulaEditorAfterClone() {
+		String userVariablename = "testvariable1";
+		String userVariablename2 = "testvariable2";
+
+		onBrickAtPosition(1).onFormulaTextField(R.id.brick_change_variable_edit_text)
+				.performNewVariable(userVariablename);
+		onBrickAtPosition(1)
+				.performCopyBrick();
+		onBrickAtPosition(1).onFormulaTextField(R.id.brick_change_variable_edit_text)
+				.performNewVariable(userVariablename2);
+		onBrickAtPosition(1).performDragNDrop(BrickCoordinatesProvider.DOWN_ONE_POSITION);
```

Comment:
what exactly is supposed to be happnening here? why is it necessary to move the brick?

Reply:
I don't really know to be honest. It was like this in the original test. Maybe there was a bug related to that at some point in the past.
Should I just remove it?

Reply:
yes, please.
since we cant guess on what is a regression test or not (since this stuff was just put in the middle of yet another test instead of creating a separate test) i think were better of to throw away stuff like this and keep tests as small as possible to not end up having the same mess again...